app/uk/gov/hmrc/alcoholdutyreturns/service/ReturnsService.scala

```diff
@@ -70,7 +73,54 @@ class ReturnsService @Inject() (
                                       returnToSubmit,
                                       returnId.appaId
                                     )
-      _                          <- EitherT.right[ErrorResponse](cacheRepository.clearUserAnswersById(returnId))
+      _                          <- EitherT.right[ErrorResponse](
+                                      cacheRepository
+                                        .get(returnId)
+                                        .map {
+                                          case Some(ua) =>
```

Comment:
no need to do the matching here

Could just do:

map { maybeUserAnswers =>
  if (maybeUserAsnwers.isEmpty) {
     logger.warn(...)
  }
  
  auditReturnSubmitted(
    maybeUserAnswers,
    ...

Reply:
yeah, it is another way of doing the same thing but it prevents code duplication so let's do the change